lib/tizen_cache.dart

```diff
@@ -144,6 +182,58 @@ class TizenEngineArtifacts extends EngineCachedArtifact {
     }
   }
 
+  Future<void> _downloadArtifactsFromAzure(
+      ArtifactUpdater updater, String buildId) async {
+    _logger.printStatus(
+        'Downloading Tizen engine artifacts from Azure Pipelines...');
+    final String downloadUrl = await _getDownloadUrlFromAzure(buildId);
+    await _downloadArtifactsFromUrl(updater, downloadUrl);
+  }
+
+  void _downloadArtifactsFromGithub(
+      OperatingSystemUtils operatingSystemUtils, String ghRunId) {
+    _logger.printStatus(
+        'Downloading Tizen engine artifacts from GitHub Actions...');
+    if (operatingSystemUtils.which('gh') == null) {
+      throwToolExit(
+        'Github CLI not found. Please install it first. '
+        'https://cli.github.com/',
+      );
+    }
+    for (final List<String> toolsDir in getBinaryDirs()) {
+      final String cacheDir = toolsDir[0];
+      final String urlPath = toolsDir[1];
+      final Directory artifactDir = location.childDirectory(cacheDir);
+      final Status status =
+          _logger.startProgress('Downloading $cacheDir tools...');
+      try {
+        if (artifactDir.existsSync()) {
+          artifactDir.deleteSync(recursive: true);
+        }
+        artifactDir.createSync(recursive: true);
+        final RunResult result = _processUtils.runSync(<String>[
+          'gh',
+          'run',
+          'download',
+          '-R',
+          'flutter-tizen/engine',
+          '-n',
+          basenameWithoutExtension(urlPath),
+          '-D',
+          artifactDir.path,
+          ghRunId,
+        ]);
+        if (result.exitCode != 0) {
+          throwToolExit(
+            'Failed to download Tizen engine artifact from Github actions. $result',
+          );
```

Comment:
It might be better to add a line break between the messages. Currently it looks like:
```
Failed to download Tizen engine artifact from Github actions. Welcome to GitHub CLI!

To authenticate, please run `gh auth login`.
```
```suggestion
          throwToolExit(
              'Failed to download Tizen engine artifact from GitHub Actions.\n$result');
```